### PR TITLE
Events test coverage

### PR DESCRIPTION
**Pull request by joshlory**
Add first integration test for "Designer" levels (where `isEventLevel` = true).

### CODE REVIEW

**Comment by joshlory** [comment]
Same approach as PR https://github.com/code-dot-org/craft/pull/47, but refactored to have both Adventurer and Designer level tests.

File: src/js/game/API/CodeOrgAPI.js [hunk 1]
```diff
@@ -65,6 +65,20 @@ module.exports.get = function (controller) {
...
     },
 
+    onEventTriggered: function (highlightCallback, type, eventType, callback) {
+      this.registerEventCallback(highlightCallback,
+        function (event) {
+          if (event.eventType !== eventType) {
+            return;
+          }
+          if (event.targetType !== type) {
+            return;
```
Comment (islemaster) [inline src/js/game/API/CodeOrgAPI.js]: Uh... `if (event.eventType !=== eventType || event.targetType !== type)`?

File: src/js/game/GameController.js [hunk 2]
```diff
@@ -219,13 +219,15 @@ class GameController {
...
 
+      tween.timeScale = 1;
       tween.start();
       tween = this.levelView.addResettableTween().to({
       }, this.timeout, Phaser.Easing.Linear.None);
 
       tween.onComplete.add(() => {
         this.endLevel(this.timeoutResult(this.levelModel));
       });
+      tween.timeScale = 1;
```
Comment (islemaster) [inline src/js/game/GameController.js]: Why is this needed?

File: test/integration/DesignerTest.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+test('Designer 1: Chicken Move (pass)', t => {
+  attempt('designer01', api => new Promise(resolve => {
+    api.onEventTriggered(null, 'chicken', 2, event => {
+      api.moveForward(null, event.targetIdentifier);
+      api.turnLeft(null, event.targetIdentifier);
+    });
+
+    api.startAttempt((success, levelModel) => {
+      debugger;
```
Comment (islemaster) [inline test/integration/DesignerTest.js]: Did you mean to leave this `debugger` statement in?  If not, can we set our linter to catch this?

Comment (joshlory) [inline test/integration/DesignerTest.js]: 😳  whoops.  It looks like eslint is only running on `/src/js` — I will fix!

File: src/js/game/GameController.js (hunk 2 above)
Comment (joshlory) [inline src/js/game/GameController.js]: Under integration tests, [the default `timeScale` is increased so tests run faster](https://github.com/code-dot-org/code-dot-org/pull/17114).

This tween is used to measure the level timeout... so I need to run it at normal speed.  In theory if all tweens are sped up by an equal amount they should still run in order, but it wasn't consistently passing.

I think when everything is run 100x there's some non-deterministic behavior of the timers (two callbacks may run in the same time slice, instead of sequentially).

Comment (islemaster) [inline src/js/game/GameController.js]: I'd suggest a comment to that effect, since in isolation (especially in src rather than test) this seems odd and it's not clear why we're doing it.

**Comment by joshlory** [comment]
@islemaster PTAL!
